### PR TITLE
Clarify differences between count() and tally()

### LINKED ISSUE #5163: Allow aggregate functions within `arrange`
Inspired by [this Stack Overflow question](https://stackoverflow.com/q/61503447/903061), perhaps `arrange` should allow aggregate functions.

Want: order groups by aggregate function

Attempt:

```
mtcars %>% 
  group_by(cyl) %>%
  arrange(mean(mpg))
# Error: incorrect size (1) at position 1, expecting : 32
```

Workaround:

```
mtcars %>%
  mutate(cyl = forcats::fct_reorder(factor(cyl), mpg, .fun = mean)) %>%
  arrange(cyl)

## or (more literally)

mtcars %>% 
  group_by(cyl) %>% 
  mutate(mean_mpg = mean(mpg)) %>%
  arrange(mean_mpg) %>%
  select(-mean_mpg)
```

Could `arrange` recycle length-one inputs up to the group size? Would make it more consistent with `filter` and `mutate`.

### LINKED ISSUE #5298: Default behavior of count() seems to have changed in 1.0.0
Noticed that the default behavior of count() has changed in 1.0.0 but wasn't listed as a breaking change in the release notes

0.8.5
``` r
library(dplyr, warn.conflicts = FALSE)
packageVersion("dplyr")
#> [1] '0.8.5'
tibble(x = 1:3, n = 3:1) %>% count(x)
#> # A tibble: 3 x 2
#>       x     n
#>   <int> <int>
#> 1     1     1
#> 2     2     1
#> 3     3     1
```
1.0.0
``` r
library(dplyr, warn.conflicts = FALSE)
packageVersion("dplyr")
#> [1] '1.0.0'
tibble(x = 1:3, n = 3:1) %>% count(x)
#> Using `n` as weighting variable
#> ℹ Quiet this message with `wt = n` or count rows with `wt = 1`
#> # A tibble: 3 x 2
#>       x     n
#>   <int> <int>
#> 1     1     3
#> 2     2     2
#> 3     3     1
```

_Originally posted by @nhols in https://github.com/tidyverse/dplyr/issues/5265#issuecomment-638496562_

### PR DESCRIPTION
**Pull request by hadley**
And revert count() to 0.8.5 behaviour. Fixes #5298

### CODE REVIEW

**Comment by yutannihilation** [comment]
> Fixes #5163

Seems a different issue? (Maybe #5298?)

**Comment by hadley** [comment]
@yutannihilation oops, yes.

**Comment by hadley** [comment]
@lionel- could you check my reasoning here please?

**Comment by hadley** [comment]
Ok, if we accidentally broke the autoguessing in 0.8.2, then lets remove it all together.

**Comment by hadley** [comment]
@yutannihilation since you have an eye for detail, would you mind taking a look at this PR. Most importantly, is the reasoning in the NEWS clear, and does it make sense to you? Thanks!

**Comment by yutannihilation** [comment]
Looks good! Just one thing to confirm, is it safe to keep using `wt = n()`? If it's a thing that the user is recommended to change, some words on this might be needed in the NEWS item.

**Comment by hadley** [comment]
Good point, probably worth keeping that check around just as a precaution.

File: NEWS.md [hunk 1]
```diff
@@ -1,7 +1,14 @@
 # dplyr (development version)
 
-* `count()` now displays the correct the hint to force a count, `wt = n()` 
-  (#5324).
+* `count()` and `tally()` no longer automatically weights by column `n` if 
+  present (#5298). dplyr 1.0.0 introduced this behaviour because of Hadley's
+  faulty memory. Historically `tally()` automatically weighted and `count()`
```
Comment (romainfrancois) [inline NEWS.md:5]: 😂

**Pull request merged**